index.d.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Decides whether or not the document is reader-able without parsing the whole thing.
+ * @return {boolean} Whether or not we suspect Readability.parse() will suceeed at returning an article object.
+ */
+export function isProbablyReaderable(
+  document: Document,
+  options?: {
+    /** The minimum node content length used to decide if the document is readerable. */
+    minContentLength?: number;
+    /** The minumum cumulated 'score' used to determine if the document is readerable. */
+    minScore?: number;
+    /** The function used to determine if a node is visible. */
+    visibilityChecker?: (node: Node) => boolean;
+  }
+): boolean;
+
+export class Readability<T = string> {
```

Comment:
I'm a little confused - what other T is supported here? Why not just not have this template-like construction and use `string` literally below?

Reply:
I'm not sure what you mean by "literally below". This makes sure that the type of `content` will be `T` (which is the return type of `serializer` and defaults to `string` if there is no serializer)

Reply:
Ah, hm. That's... interesting. @danburzo is there a particular reason we shouldn't require it to be a string (even if it can be a non-string right now)? Feels odd to have something called "serializer" that then does other transforms and produces something other than a string.

Reply:
> is there a particular reason we shouldn't require it to be a string (even if it can be a non-string right now)? 

We introduced `serializer` to support other types of DOM serialization (e.g. XHTML/XML for EPUB, #605). In my particular case, where Readability is part of a larger processing pipeline, I found it's actually more efficient to skip an entire serialization/parse pair altogether:

```js
const R = new Readability(dom.window.document, {
        // ...
	/*
		Change Readability's serialization to return 
		a DOM element (instead of a HTML string) 
		as the `.content` property returned from `.parse()`

		This makes it easier for us to run subsequent
		transformations (sanitization, hyphenation, etc.)
		without having to parse/serialize the HTML repeatedly.
	 */
	serializer: el => el
});
```

In my previous comment, I had suggested the change in the TypeScript definition for the sake of accuracy, but if this detail of the `serializer` option obscures the class as a whole, I think we can safely pretend it always returns a string here?

Reply:
Hm. Sorry for not doing a good job at reading before. No-op seems more in keeping (like saying somehow "don't serialize this at all"). We could make it return `Node | string`, assuming the typescript definition file allows for that? Then it won't require parametrization/templatization of the class. We should probably clarify this in the jsdoc and/or docs though.

Reply:
So it's a bit hard to to have the return type of `parse` depends on the return type of `serializer` without having it in a template. IMO it's more important to have an exact return type for `parse` rather than for `serializer`.

The template is transparent for the user, they won't ever need to use it so I don't think it's obscuring the class too much?